benchmark/scenarios/test/test_rate_measurer.py

```diff
@@ -0,0 +1,147 @@
+from flocker.testtools import TestCase
+
+from .._rate_measurer import RateMeasurer
+
+
+class RateMeasurerTest(TestCase):
+    """
+    RateMeasurer tests.
+    """
+
+    def send_requests(self, rate_measurer, num_requests, num_samples):
+        """
+        Helper function that will send the desired number of request.
+
+        :param rate_measurer: The `RateMeasurer` we are testing.
+        :param num_requests: The number of request we want to send.
+        :param num_samples: The number of samples to collect.
+        """
+        for i in range(num_samples * num_requests):
+            rate_measurer.request_sent()
+
+    def receive_requests(self, rate_measurer, num_requests, num_samples):
+        """
+        Helper function that will receive the desired number of requests.
+
+        :param rate_measurer: The `RateMeasurer` we are testing.
+        :param num_requests: The number of request we want to receive.
+        :param num_samples: The number of samples to collect.
+        """
+        ignored = ""
+        for i in range(num_samples):
+            for i in range(num_requests):
+                rate_measurer.response_received(ignored)
+            rate_measurer.update_rate()
+
+    def failed_requests(self, rate_measurer, num_failures, num_samples):
+        """
+        Helper function that will result the desired number of response
+        failures.
+
+        :param rate_measurer: The `RateMeasurer` we are testing.
+        :param num_failures: The number of requests we want to fail.
+        :param num_samples: The number of samples to collect.
+        """
+        result = None
+        for i in range(num_samples):
+            for i in range(num_failures):
+                rate_measurer.request_failed(result)
+            rate_measurer.update_rate()
+
+    def increase_rate(self, rate_measurer, num_requests, num_samples):
+        """
+        Helper function that will increase the rate, sending the
+        desired number of request, and receiving the same
+        amount of them.
+
+        :param rate_measurer: The `RateMeasurer` we are testing.
+        :param num_requests: The number of request we want to make.
+        :param num_samples: The number of samples to collect.
+        """
+        self.send_requests(rate_measurer, num_requests, num_samples)
+        self.receive_requests(rate_measurer, num_requests, num_samples)
+
+    def test_rate_is_zero_when_no_samples(self):
+        """
+        When no samples have been collected, the rate should be 0.
```

Comment:
Write test method docstrings without "should" - eg "When no samples have been collected, the rate is 0."